tools/authors.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+help() {
+	echo "Usage:"
+	echo "  $0 [opts]"
+	echo ""
+	echo "  Generate an alphabetically ordered list of authors and update the authors file"
+	echo ""
+	echo "  Options:"
+	echo "    -c		Checks the authors file is valid without modifying it."
+	echo "    -h		Show this help."
+}
+
+while getopts "ch" opt; do
+	case $opt in
+	c) check_only=1 ;;
+	*) help; exit ;;
+	esac
+done
+shift $(($OPTIND - 1))
+
+authors_file=$(git rev-parse --show-toplevel)/AUTHORS.md
+authors_list="$(git log --format='%aN' | grep -v dependabot | sort -u)"
```

Comment:
a nit that doesn't really change anything (doesn't happen for sure at the moment):

`--no-merges` is theoretically needed here (maintainer that only merges)

other kind of contributions (still valuable) should be recognized in a separate section

Reply:
IMO that doesn't make an "author". But yes, merging PRs is very valuable :) If we want to recognize this we can have a static list of maintainers; as you suggested, in a separate section.

Reply:
> IMO that doesn't make an "author". But yes, merging PRs is very valuable :) If we want to recognize this we can have a static list of maintainers; as you suggested, in a separate section.

agreed, I was suggesting `--no-merges` for that reason

Reply:
Oh I see sorry. I was absolutely certain the above was done with `--no-merges`; all my local tests were. No idea how this got dropped but that is definitively needed. Thanks!

Reply:
Or maybe I only used it when testing ordering by # of commits... Anyway, thanks for spotting this :)

Reply:
well, the result is the same :)